mdata/cache/ccache.go

```diff
@@ -31,10 +32,10 @@ type CCache struct {
 	sync.RWMutex
 
 	// one CCacheMetric struct per metric key, indexed by the key
-	metricCache map[string]*CCacheMetric
+	metricCache map[schema.AMKey]*CCacheMetric
 
 	// sets of metric keys, indexed by their raw metric keys
-	metricRawKeys map[string]map[string]struct{}
+	metricRawKeys map[schema.AMKey]map[schema.AMKey]struct{}
```

Comment:
could this be optimized into something like `map[schema.AMKey]map[schema.Archive]`?